CONTRIBUTING.md

```diff
@@ -20,7 +20,9 @@ Wesnoth's engine conforms to the C++17 standard. We encourage the use of standar
 
 All C++, WML and Lua files are in UTF-8, as we use Gettext-style translations, and translatable strings use some punctuation that's outside of the ASCII subset. More details are in the [Typography Style Guide](https://wiki.wesnoth.org/Typography_Style_Guide) and the guide to [using Gettext strings](https://wiki.wesnoth.org/GettextForWesnothDevelopers).
 
-If your pull request touches the engine's C++ source code, we recommend (but don't require) you run `clang-format` on your changes before submission (Visual Studio Code gives you a handy context menu option to do so). This ensures that your code remains formatted according to our conventions. Make a local commit before running `clang-format`, in case more code than expected gets changed.
+WML files must be formatted with wmlindent, as this will be checked by the CI. Currently the data/gui and data/schema directories are exceptions to this, some of the unit tests are exceptions too.
+
+For C++ files, we have a configuration file for `clang-format`, however C++ formatting is handled by comments in code review rather than enforcement by CI. Some GUIs give a handy context menu to use it, some such as Visual Studio Code are able to run it on selected text rather than the full file. We've considered a full reformat of the codebase, and decided against it; for C++ changes in files that aren't matching the standard it's left to the judgement of the contributor. We recommend making a local commit before running `clang-format`, in case more code than expected gets changed.
```

Comment:
I would say enforcement by CI is not done _yet_.

Regarding a full reformat of the code base IMO we decided against doing it in one go. We should strive to do it incrementally. Which IMO would, as a vital step, include a CI check to enforce formatting of what is committed. (https://github.com/llvm/llvm-project/blob/main/clang/tools/clang-format/clang-format-diff.py)

The text here sounds to me a bit too much like we don't ever want to reach a state where our code base and clang-format agree.